src/generics/middleware/authenticator.js

```diff
@@ -37,7 +38,21 @@ module.exports = async function (req, res, next, token = '') {
 		delete req.headers[e]
 	})
 
-	var token = req.headers['x-auth-token']
+	// Check if a Bearer token is required for authentication
+	let authHeader = req.headers['x-auth-token']
+	if (isBearerRequired) {
+		const [authType, extractedToken] = authHeader.split(' ')
+		if (authType.toLowerCase() !== 'bearer') {
+			rspObj.errCode = CONSTANTS.apiResponses.TOKEN_INVALID_CODE
+			rspObj.errMsg = CONSTANTS.apiResponses.TOKEN_INVALID_MESSAGE
+			rspObj.responseCode = HTTP_STATUS_CODE['unauthorized'].status
+			return res.status(HTTP_STATUS_CODE['unauthorized'].status).send(respUtil(rspObj))
+		}
+		token = extractedToken?.trim()
```

Comment:
@priyanka-TL where are we defining this token variable?

Reply:
Its already defined in function argument, so if we tried to define again it cause issue

Reply:
@priyanka-TL okey got it. thanks